tests/test_asg.py

```diff
@@ -832,6 +832,31 @@ def test_valid_asg_with_launch_templates(self):
         resources = p.run()
         self.assertEqual(len(resources), 1)
 
+    def test_asg_invalid_filter_validation(self):
+        """Validate policy execution mode
+
+        The valid/invalid filters should cause Lambda mode policies to fail, but
+        pass for other execution modes whether specified explicitly or not.
+        """
+        base_policy = {
+            "name": "asg-invalid-filter",
+            "resource": "asg",
+            "filters": ["invalid"],
+        }
+
+        p = self.load_policy(base_policy)
+        p.validate()
+        p = self.load_policy({**base_policy, "mode": {"type": "pull"}})
```

Comment:
I'd just leave a comment here about why you are doing this.

Reply:
Hmm I had a line comment initially but added that bit to the docstring instead. This suggests I should put it back :) - done